getMultiStarts.m

```diff
@@ -222,40 +234,51 @@
         error_count = 0;
         
         % Test evaluation of objective function at starting point
+        % Only for multi-start local, since other optimizers use a 
+        % different initialization
         if (strcmp(options.localOptimizer, 'fmincon'))
             if (strcmp(options.localOptimizerOptions.Hessian, 'on'))
-                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,i)); % objectiveWrapWErrorCount(parameters.MS.par0(:,i),objective_function,options.obj_type,options.objOutNumber);
+                % We have to check Hessian AND gradient computation, since
+                % simulation can give different results if we're close to
+                % numerical failure
+                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,iMS));
```

Comment:
Ah yes, that makes sense.

I think we only need to ensure here that the first call from `fmincon` should not fail. This is probably different for `interior-point` and `trust-region-reflective`. `interior-point` has this additional handle for the Hessian and will thus call `negLogPostWErrorCount` with 2 output arguments and `trust-region` will call it with 3 output arguments.

If it is really case that negLogPostWErrorCount provide different results for J_0 with 2 and 3 output arguments I do not see how taking the max over both values does any good. It might fix the problem in the particular manifestation that you encountered, but I do not see how this can generalise.

I think if the evaluation of the objective function is numerically so fragile, we should not hide this fact from the user, but rather throw an error because it is highly unlikely that the optimization will yield anything reasonable. 

Yet, I still don't like the computation overhead from calling the objective function multiple times at initialisation. For complex models this can easily mean wasting multiple hourse of computation time and which is especially cumbersome when debugging the optimization.